src/test/kotlin/app/batch/KeyPairGeneratorTest.kt

```diff
@@ -0,0 +1,142 @@
+                "adb.collection.0000.json.gz.encryption.json", "cdb.collection.0000.json.gz.encryption.json",
+                "sdb.collection.0000.json.gz.encryption.json", "adb.collection.0000.json.gz.pdf", "adb.collection.0000.json.gz.docx")
+
+        val actualKeyPairs = keyPairGenerator.generateKeyPairs(keys, fileFormatRegex, dataFileExtensionRegex, metadataFileExtensionRegex)
+
+        val captor = argumentCaptor<ILoggingEvent>()
+        verify(mockAppender, times(8)).doAppend(captor.capture())
+        val formattedMessages = captor.allValues.map { it.formattedMessage }
+        assertTrue(formattedMessages.contains("adb.collection.0000.json.gz.pdf,adb.collection.0000.json.gz.docx matched file format but not matched neither data nor metadata file extension"))
+        assertEquals(expectedKeyPairs, actualKeyPairs)
+
+    }
+
+    @Test
+    fun Should_Log_When_Data_File_Doesnt_But_Metadata_File_Exist() {
+
+        val root = LoggerFactory.getLogger(ch.qos.logback.classic.Logger.ROOT_LOGGER_NAME) as ch.qos.logback.classic.Logger
+        val mockAppender: Appender<ILoggingEvent> = mock()
+        root.addAppender(mockAppender)
+        val expectedKeyPairs = listOf(KeyPair("cdb.collection.0000.json.gz.enc", "cdb.collection.0000.json.gz.encryption.json"),
+                KeyPair("sdb.collection.0000.json.gz.enc", "sdb.collection.0000.json.gz.encryption.json"))
+
+        val keys = listOf("cdb.collection.0000.json.gz.enc", "sdb.collection.0000.json.gz.enc",
+                "adb.collection.0000.json.gz.encryption.json", "cdb.collection.0000.json.gz.encryption.json",
+                "sdb.collection.0000.json.gz.encryption.json")
+
+        val actualKeyPairs = keyPairGenerator.generateKeyPairs(keys, fileFormatRegex, dataFileExtensionRegex, metadataFileExtensionRegex)
+
+        val captor = argumentCaptor<ILoggingEvent>()
+        verify(mockAppender, times(9)).doAppend(captor.capture())
+        val formattedMessages = captor.allValues.map { it.formattedMessage }
+        assertTrue(formattedMessages.contains("Data file not found for the metadata file adb.collection.0000.json.gz.encryption.json"))
+        assertEquals(expectedKeyPairs, actualKeyPairs)
+
+    }
+
+    @Test(expected = RuntimeException::class)
+    fun Should_Throw_Exception_When_Metadata_File_Doesnt_Exist() {
+
+        listOf(KeyPair("adb.collection.0000.json.gz.enc", null),
+                KeyPair("cdb.collection.0000.json.gz.enc", "cdb.collection.0000.json.gz.encryption.json"),
+                KeyPair("sdb.collection.0000.json.gz.enc", "sdb.collection.0000.json.gz.encryption.json"))
+
+        val keys = listOf("adb.collection.0000.json.gz.enc", "cdb.collection.0000.json.gz.enc", "sdb.collection.0000.json.gz.enc",
+                "cdb.collection.0000.json.gz.encryption.json",
+                "sdb.collection.0000.json.gz.encryption.json")
+
+        keyPairGenerator.generateKeyPairs(keys, fileFormatRegex, dataFileExtensionRegex, metadataFileExtensionRegex)
```

Comment:
Can we please assert the exception message, to be sure we are not leaking data, and what will be in our logs is useful?

Reply:
Asserted the log message